File: app/src/main/java/ng/max/binger/data/ApiService.java
package ng.max.binger.data;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {
    @GET("tv/airing_today")
    Observable<TvShowResponse> getTVAiringToday(@Query("api_key") String apiKey);

    @GET("tv/popular")
    Observable<TvShowResponse> getPopularMovies(@Query("api_key") String apiKey);

    @GET("tv/{id}")
    Observable<TvShow> getMovieDetails(@Path("id") int id, @Query("api_key") String apiKey);
}

File: app/src/main/java/ng/max/binger/data/ApiService.kt
package ng.max.binger.data

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiService {
    @GET("tv/airing_today")
    fun getTVAiringToday(@Query("api_key") apiKey: String): Observable<TvShowResponse>

    @GET("tv/popular")
    fun getPopularMovies(@Query("api_key") apiKey: String): Observable<TvShowResponse>

    @GET("tv/{id}")
    fun getMovieDetails(@Path("id") id: Int, @Query("api_key") apiKey: String): Observable<TvShow>
}